src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs

```diff
@@ -39,14 +40,20 @@ private void RoundTrip(Action<ObjectWriter> writeAction, Action<ObjectReader> re
         {
             using var stream = new MemoryStream();
 
-            using (var writer = new ObjectWriter(stream, leaveOpen: true))
+            ImmutableArray<object> keepAliveObjects;
+            using (var writer = new ObjectWriter(stream, leaveOpen: true, canKeepObjectsAlive: true))
             {
                 writeAction(writer);
+
+                keepAliveObjects = writer.TakeKeepAliveObjects();
             }
 
             stream.Position = 0;
```

Comment:
do we need some sort of explicit GC action here to try to purge data to ensure that the keep-aliving is working?

Reply:
Tests of this form are historically extremely unreliable. Also the keep alive functionality is the caller's responsibility, so covering it in test would not reflect in any way on the production behavior.

Reply:
I could add _targeted_ tests for the API behavior of `ObjectWriter.KeepAlive` and `TakeKeepAliveObjects`. These would not need rely on any specific GC behavior.

Reply:
> Tests of this form are historically extremely unreliable.

Reply:
I already implemented the tests locally (there is no need to use GC APIs for it). I'm working on docs before updating the PR.

Reply:
➡️ Now tested